src/AlertBar/index.js

```diff
@@ -0,0 +1,112 @@
+import React, { PureComponent } from 'react'
+import propTypes from 'prop-types'
+import cx from 'classnames'
+import styles, { ANIMATION_TIME } from './styles'
+
+import { Actions, actionsPropType } from './Actions'
+import { Dismiss } from './Dismiss'
+import { Icon, iconPropType } from './Icon'
+import { Message } from './Message'
+
+class AlertBar extends PureComponent {
+    state = {
+        visible: false,
+    }
+
+    componentDidMount() {
+        this.startTime = Date.now()
+        this.timeRemaining = this.props.duration
+        this.startDisplayTimeout()
+        this.show()
+    }
+
+    componentWillUnmount() {
+        clearTimeout(this.displayTimeout)
+        clearTimeout(this.onHiddenTimeout)
+    }
+
+    startDisplayTimeout = () => {
+        this.displayTimeout = setTimeout(() => {
+            this.hide()
+        }, this.timeRemaining)
+    }
+
+    stopDisplayTimeOut = () => {
+        this.timeRemaining = this.timeRemaining - (Date.now() - this.startTime)
+        clearTimeout(this.displayTimeout)
+    }
+
+    hide = () => {
+        clearTimeout(this.displayTimeout)
+        this.setState({ visible: false })
+
+        if (this.props.onHidden) {
+            this.onHiddenTimeout = setTimeout(
+                this.props.onHidden,
+                ANIMATION_TIME
+            )
+        }
+    }
+
+    show() {
+        setTimeout(() => {
```

Comment:
Why use `setTimeout` here? Wouldn't `requestAnimationFrame` be more appropriate?

Reply:
Yes, good suggestion.